### PR TITLE
feat: Drawer add loading prop to show spinner

### PR DESCRIPTION
**Pull request by Enigama**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
It would be nice to have loading prop, it will solve some issue when async data loads in drawer and we don't want to show drawer without title or some body parts.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
1. `feature` branch please.
2. Should drawer content be prevented from rendering while loading? As far as I know spin doesn't block content.

**Comment by Enigama** [comment]
> 1. `feature` branch please.
> 
> 2. Should drawer content be prevented from rendering while loading? As far as I know spin doesn't block content.

1 - sorry, will change branch
2 - we could just hide it by adding some additional style to spinner however if it necessary can try to block content while loading is true

File: components/drawer/demo/loading.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      <Button type="primary" onClick={showDrawer}>
+        Open
+      </Button>
+      <Drawer
+        title="Basic Drawer"
+        placement="right"
+        closable={false}
+        onClose={onClose}
+        open={open}
+        loading={loading}
```
Comment (afc163) [inline components/drawer/demo/loading.tsx:28]: How about adding a nprogress loading UI for Drawer?

https://ricostacruz.com/nprogress/

```jsx
loading: boolean | { progress?: number };
```

Comment (Enigama) [reply components/drawer/demo/loading.tsx:28]: @yoyo837 in my experience spinner look more appropriate.

Comment (Enigama) [reply components/drawer/demo/loading.tsx:28]: I could play with, one question, does this library [nprogress](https://ricostacruz.com/nprogress/) already installed?

**Comment by Enigama** [comment]
I should probably close this PR and open a new one since my branch is based on `master` and then I load all from `feature` branch.

**Pull request closed without merging**